perfcounters/win_perf_counters.go

```diff
@@ -4,11 +4,69 @@ package perfcounters
 
 import (
 	"fmt"
+	"log"
 	"strings"
+	"time"
+
+	"github.com/pkg/errors"
 )
 
 type WinPerfCountersWatcher struct {
-	query PerformanceQuery
+	query        PerformanceQuery
+	ongoingQuery PerformanceQuery
+	handles      map[string]PDH_HCOUNTER
+}
+
+func (m *WinPerfCountersWatcher) StartQuery(counterPath string, interval time.Duration) error {
```

Comment:
> I think it can be more clear if StartQuery will be non-struct func and returns WinPerfCountersWatcher
The function accesses the member variable `ongoingQuery` so the function is coupled to the `WinPerfCountersWatcher`. Don't see the benefit yet why it should return a watcher. The watcher holds multiple queries and handles and to start a continuous one is an action of the watcher which works with the underlying resources.

Reply:
> Better naming can be StartContinuousQuery otherwise it is no clear naming difference between Start() and StartQuery()

✔️